### PR TITLE
Refactor: Use foundry strucure and add tests

### PR DESCRIPTION
**Pull request by mathdroid**
- moved contracts to src/
- add simple tests

image

### CODE REVIEW

File: test/ERC404.t.sol [hunk 1]
```diff
@@ -0,0 +1,33 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import "../src/contracts/examples/ExampleERC404.sol";
+
+contract Erc404Test is Test {
+    ExampleERC404 public pandora;
```
Comment (0xacme) [inline test/ERC404.t.sol]: Remove pandora references here, lets keep it as close to generalized as possible

File: script/Counter.s.sol [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
```
Comment (0xacme) [inline script/Counter.s.sol]: Imo should kill the script dir and Counter, given unused.

**Review by 0xacme — COMMENTED** [review]
Lets adjust dir structure such that we don't have "contracts" anymore. Ex. src/contracts/examples should just be src/examples.

You'll need to update Hardhat source config most likely, but make sure to double check that `pnpm hardhat test` runs as expected still.

Below is an example of what you'll want to change in hardhat config.

```
  paths: {
    sources: "./contracts",
 ```

**Comment by mathdroid** [comment]
> Lets adjust dir structure such that we don't have "contracts" anymore. Ex. src/contracts/examples should just be src/examples.
> 
> You'll need to update Hardhat source config most likely, but make sure to double check that `pnpm hardhat test` runs as expected still.
> 
> Below is an example of what you'll want to change in hardhat config.
> 
> ```
>   paths: {
>     sources: "./contracts",
> ```

image

image

unfolded contracts dir, both tests run ok

**Pull request merged**